src/ai/backend/agent/server.py

```diff
@@ -195,6 +200,33 @@ async def _inner(self_: AgentRPCServer, request: RPCMessage) -> Any:
         return _inner
 
 
+def _collect_metrics(observer: RPCMetricObserver) -> Callable:
+    def decorator(meth: Callable) -> Callable[[AgentRPCServer, RPCMessage], Any]:
+        @functools.wraps(meth)
+        async def _inner(self: AgentRPCServer, *args, **kwargs) -> Any:
```

Comment:
We can apply more strict type hints here but then we have to update the type hints of `RPCFunctionRegistry.__call__()`.
Let's leave it as a minor issue